### ISSUE TITLE
about mediapipe

### ISSUE THREAD

**Issue by zcswdt** [rlee3359/folding-by-hand#5]
I found that when I used gesture recognition, the recognition was inaccurate and even failed to track my hands, so it was difficult to complete a complete set of actions. Have you encountered this problem?

**Comment by rlee3359** [comment]
The performance of the tracking can vary based on different factors, including what camera you use, lighting, etc.
There are several parameters you can tweak, such as confidence and tracking vs re-detection on each frame.
https://github.com/rlee3359/folding-by-hand/blob/869e4695043e524013213e93e55462e960cf426c/collect_demos.py#L147C1-L147C24

If you have no luck with that I would try the minimal demo code running on your camera and then go from there:
https://ai.google.dev/edge/mediapipe/solutions/vision/hand_landmarker

**Comment by zcswdt** [comment]
I am using a realsense L515 camera. What kind of environment do I need for lighting? Do you need to be under direct light or avoid light sources. I adjusted the complexity of the gesture detection model to 2, but an error occurred. At present, it is unclear what the reason is. I will try again tomorrow to test the confidence parameters, but I have tried many times. The opponent's tracking is intermittent and not continuous, and the finger detection is also intermittent.

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/12/2024 21:27 |
| To | rlee3359/folding-by-hand ***@***.***> |
| Cc | zcswdt ***@***.***>,
Author ***@***.***> |
| Subject | Re: [rlee3359/folding-by-hand] about mediapipe (Issue #5) |

The performance of the tracking can vary based on different factors, including what camera you use, lighting, etc.
There are several parameters you can tweak, such as confidence and tracking vs re-detection on each frame.
https://github.com/rlee3359/folding-by-hand/blob/869e4695043e524013213e93e55462e960cf426c/collect_demos.py#L147C1-L147C24

If you have no luck with that I would try the minimal demo code running on your camera and then go from there:
https://ai.google.dev/edge/mediapipe/solutions/vision/hand_landmarker

**Comment by rlee3359** [comment]
Can you try testing this web demo with your L515 camera?
https://mediapipe-studio.webapps.google.com/demo/hand_landmarker 

You don't need any sophisticated lighting, it just needs to be bright enough to clearly see the hands against a contrasting background.

If you add this visualization code back in you can see the whole hand detection:
https://github.com/rlee3359/folding-by-hand/blob/869e4695043e524013213e93e55462e960cf426c/collect_demos.py#L241

**Comment by zcswdt** [comment]
> You don't need any sophisticated lighting, it just needs to be bright enough to clearly see the hands against a contrasting background.
> 
> If you add this visualization code back in you can see the whole hand detection:
> 
> https://github.com/rlee3359/folding-by-hand/blob/869e4695043e524013213e93e55462e960cf426c/collect_demos.py#L241

In the same environment, I found that the link you gave me has very good results for gesture tracking and detection, while the one I installed using pip has very poor tracking and detection results. Yes, I visualized the code you commented and found that the detection effect is very poor.

**Comment by rlee3359** [comment]
I see, that's good news since it means the hand tracking model should work well. I'm not sure what the issue could be since it should perform similarly to the browser based model. I recommend creating a standalone app  based on the example code and tweaking it till it works reliably and then moving back to the data collection and applying those changes.

**Comment by zcswdt** [comment]
> I see, that's good news since it means the hand tracking model should work well. I'm not sure what the issue could be since it should perform similarly to the browser based model. I recommend creating a standalone app based on the example code and tweaking it till it works reliably and then moving back to the data collection and applying those changes.

I tested the gesture tracking code separately and found that the problem seems to be in the static_image_mode parameter. I understand that if it is true, it is used to process images, and if it is false, it is used to process video streams. I set it to false and the effect is significantly improved, but there are still some minor problems, but it should not affect data collection. Have you understood Yolov8? I see that it can be used for gesture tracking and key point detection, I don't know how effective it is?
`with mp_hands.Hands(max_num_hands=1, model_complexity=1, min_detection_confidence=0.5, min_tracking_confidence=0.5,
                    static_image_mode=False) as hands:`
